openupgrade_scripts/scripts/hr/16.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Coop IT Easy (https://coopiteasy.be)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+_moved_fields = [
+    "__last_update",
+    "_order",
+    "display_name",
+    "name",
+]
+
+_xmlid_renames = [
+    (
+        "hr_contract.access_hr_contract_type_manager",
+        "hr.access_hr_contract_type_manager",
+    ),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.update_module_moved_models(
+        env.cr, "hr.contract.type", "hr_contract", "hr"
+    )
+    openupgrade.update_module_moved_fields(
+        env.cr, "hr.contract.type", _moved_fields, "hr_contract", "hr"
+    )
+    openupgrade.rename_xmlids(env.cr, _xmlid_renames)
+    # Backup Many2many relation between hr.plan and hr.plan.activity.type
+    openupgrade.remove_tables_fks(env.cr, ["hr_plan_hr_plan_activity_type_rel"])
+    openupgrade.rename_tables(env.cr, [("hr_plan_hr_plan_activity_type_rel", None)])
```

Comment:
Hmmm, if I comment the line it says the table doesn't exist, investigating

Reply:
I suspect this is an issue local to your computer.

Heads-up that the referenced code is being moved to https://github.com/OCA/openupgradelib/pull/343

Reply:
Hi @remytms @carmenbianca I have the same issue here !
Found out that replacing the None by a dummy name on line 32 :
`openupgrade.rename_tables(env.cr, [("hr_plan_hr_plan_activity_type_rel", 'test_hr_plan_rel')])`

(and then replacing the `openupgrade.get_legacy_name` occurences in post_migration.py with my dummy name)

It works properly.
I am wondering if there is not a limit with the number of characters a database table name can support ? (I am using Postgres 13 here)

Reply:
@remi-filament is right. It has to be with the maximum number of character for the name of a field in PostgreSQL.

I had to search the PostgreSQL manuel, but it's [here](https://www.postgresql.org/docs/14/datatype-character.html#DATATYPE-CHARACTER-SPECIAL-TABLE). 63 ASCII character long for all the names of the table, fields, etc.

If name specified is to long, PostgreSQL cut it at 63 characters. When creating default name for indices, constrains, etc it use a [dedicated algorithm](https://doxygen.postgresql.org/indexcmds_8c.html#a04ac525d457328bbe04c0f5924f761f7), which consist of cropping the longest element of the name until the whole name fits into 63 characters.

In this case, we use `get_legacy_name()` to retrieve a new name for the table. `get_legacy_name()` add something like `openupgrade_legacy_16_0_...`. Also the `rename_table()` function also tries to rename the indices, constrains, etc.

The original indices name for this table are:
```
Indexes:
    "hr_plan_hr_plan_activity_type_rel_pkey" PRIMARY KEY, btree (hr_plan_id, hr_plan_activity_type_id)
    "hr_plan_hr_plan_activity_type_hr_plan_activity_type_id_hr_p_idx" btree (hr_plan_activity_type_id, hr_plan_id)
Foreign-key constraints:
    "hr_plan_hr_plan_activity_type_rel_hr_plan_activity_type_id_fkey" FOREIGN KEY (hr_plan_activity_type_id) REFERENCES hr_plan_activity_type(id) ON DELETE CASCADE
    "hr_plan_hr_plan_activity_type_rel_hr_plan_id_fkey" FOREIGN KEY (hr_plan_id) REFERENCES hr_plan(id) ON DELETE CASCADE
```

Which gives after the rename and the rename with `get_legacy_name()`:
- `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_rel` (57 char)
- `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_rel_pkey` (62 char)
- `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_hr_plan_activity_type_id_hr_p_idx` (87 char) -> `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_hr_plan_a` (63 char).
- `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_rel_hr_plan_activity_type_id_fkey` (87 char) -> `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_rel_hr_pl` (63 char)
- `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_rel_hr_plan_id_fkey` (73 char) -> `openupgrade_legacy_16_0_hr_plan_hr_plan_activity_type_rel_hr_pl` (63 char)

We see that two constrains are renamed the same way, which is a problem.

I think that this case should be fixed directly on `openupgradelib`. I will suggest something.

In this script I will set the new name of the table instead of using `get_legacy_name()`.

Reply:
Hi @remytms did you progressed on this fix?

Reply:
Rémy is out of office for 2 weeks :-)

Reply:
@marielejeune I've not proposed fixes on `openupgradelib` yet, but in this migration script I use a custom name for the table. So that it should not be an issue anymore.